UndertaleModLib/UndertaleLists.cs

```diff
@@ -7,171 +7,214 @@
 
 namespace UndertaleModLib
 {
-    public abstract class UndertaleListBase<T> : ObservableCollection<T>, UndertaleObject
+    /// <summary>
+    /// Basic observable list type, usable for data models to be presented in interfaces.
+    /// </summary>
+    public class UndertaleObservableList<T> : ObservableCollection<T>
     {
+        // Reference to the private internal list inside Collection<T>
         private readonly List<T> internalList;
 
-        public UndertaleListBase()
-        {
-            try
-            {
-                FieldInfo itemsField = typeof(Collection<T>)
-                                       .GetField("items", BindingFlags.NonPublic | BindingFlags.Instance);
-                internalList = (List<T>)itemsField.GetValue(this);
+        // Internal list field (name is guaranteed not to change due to the underlying type being serializable)
```

Comment:
i dont quite get what the stuff within parantheses is supposed to mean.

Reply:
Reworded it now to be a bit more clear.